feedr-starter-code/js/app.js

```diff
@@ -1,3 +1,115 @@
-/*
-  Please add all Javascript code to this file.
-*/
\ No newline at end of file
+// this class for the apis
+//Note : if we want to add a new api we will creat a new function like the function api1()
+class Apis{
+  constructor() {
+    this.newsSources = ['ign','the-verge','the-wall-street-journal','polygon','national-geographic'];
+  }
+  // this function to creat one api
+  api1(){
+    let url = 'https://newsapi.org/v2/everything?';
+    let searchOptions = {
+      sources: getUrlFilter(),
+      from: '2019-04-01',
+      language: 'en',
+      apikey: '64ee82ee9a00407ca078ef42a707726d',
```

Comment:
It's better if you put the API key in an external file and not push it to Github!